### PR TITLE
create Announcement and get categories

### PR DESCRIPTION
**Pull request by SenkoVlad**

### CODE REVIEW

File: UAE.Application/Services/Implementations/AnnouncementService.cs [hunk 1]
```diff
@@ -3,18 +3,43 @@
...
+        var userId = _userService.GetCurrentUserId();
+
+        if (string.IsNullOrWhiteSpace(userId))
+        {
+            return new OperationResult(ResultMessage: "userId is not set in cookies", IsSucceed: true);
+        }
+        
+        var announcement = ApplicationMapper.Mapper.Map<Announcement>(createAnnouncementModel);
+        announcement.User.ID = userId;
+        announcement.CreatedDateTime = DateTime.Now;
```
Comment (buhaiov-vs) [inline UAE.Application/Services/Implementations/AnnouncementService.cs:38]: UtcNow

File: UAE.Api/Validations/ValidationRules/CreateAnnouncementModelValidator.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+using UAE.Core.Entities;
+
+namespace UAE.Api.Validations.ValidationRules;
+
+public class CreateAnnouncementModelValidator : AbstractValidator<CreateAnnouncementModel>
+{
+    public CreateAnnouncementModelValidator()
+    {
+        RuleFor(c => c.Title)
+            .NotNull()
```
Comment (buhaiov-vs) [inline UAE.Api/Validations/ValidationRules/CreateAnnouncementModelValidator.cs:12]: NotNull Already handled by non nullable field

File: UAE.Application/Services/Implementations/TokenService.cs [hunk 3]
```diff
@@ -69,7 +70,7 @@ public string CreateToken(User user)
             claims: claims,
             audience: _settings.Value.Jwt.Issuer,
             issuer: _settings.Value.Jwt.Issuer,
-            expires: DateTime.Now.AddDays(1),
+            expires: DateTime.Now.AddDays(10),
```
Comment (buhaiov-vs) [inline UAE.Application/Services/Implementations/TokenService.cs:73]: UtcNow

File: UAE.Infrastructure/UAE.Infrastructure.csproj [hunk 4]
```diff
@@ -15,4 +15,10 @@
       <PackageReference Include="MongoDB.Entities" Version="21.0.1" />
     </ItemGroup>
 
+    <ItemGroup>
+      <Reference Include="Microsoft.AspNetCore.Http.Abstractions">
+        <HintPath>C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App\7.0.0\Microsoft.AspNetCore.Http.Abstractions.dll</HintPath>
```
Comment (buhaiov-vs) [inline UAE.Infrastructure/UAE.Infrastructure.csproj:20]: absolute path, let's change to relative

**Review by buhaiov-vs — APPROVED** [review]

File: UAE.Application/Services/Implementations/AnnouncementService.cs (hunk 1 above)
Comment (buhaiov-vs) [reply UAE.Application/Services/Implementations/AnnouncementService.cs]: still valid

Comment (SenkoVlad) [reply UAE.Application/Services/Implementations/AnnouncementService.cs]: done

File: UAE.Infrastructure/UAE.Infrastructure.csproj (hunk 4 above)
Comment (SenkoVlad) [reply UAE.Infrastructure/UAE.Infrastructure.csproj:20]: just missed

**Pull request merged**